homeassistant/util/async_.py

```diff
@@ -30,20 +30,6 @@ def asyncio_run(main: Awaitable[_T], *, debug: bool = False) -> _T:
             loop.close()
 
 
-def run_coroutine_threadsafe(
-    coro: Union[Coroutine, Generator], loop: AbstractEventLoop
-) -> concurrent.futures.Future:
-    """Submit a coroutine object to a given event loop.
-
-    Return a concurrent.futures.Future to access the result.
-    """
-    ident = loop.__dict__.get("_thread_ident")
```

Comment:
We will lose this check. I guess it's ok because we have migrated the whole core to asyncio

Reply:
yep

Reply:
oh perfect.

Reply:
Did you see if we use `_thread_ident` still or can we remove it ?

Reply:
Ident, we can remove it. on all our function we call `loop.call_soon_threadsafe(run_callback)` which is guard by asyncio

Reply:
We need it currently for logging. I will rewrite this abstraction to next beta, so we can remove that guards complete. For now, it should be fine